src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2429,4 +2442,345 @@ public Response getTimestamps(@PathParam("identifier") String id) {
+                                newFilename,
+                                newFileContentType,
+                                newStorageIdentifier,
+                                null,
+                                optionalFileParams, true);
+
+                        if (addFileHelper.hasError()) {
+
+                            JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                    .add("storageIdentifier ", newStorageIdentifier)
+                                    .add("error Code: ", addFileHelper.getHttpErrorCode().toString())
+                                    .add("message ", addFileHelper.getErrorMessagesAsString("\n"));
+
+                            jarr.add(fileoutput);
+
+                        } else {
+                            String successMsg = BundleUtil.getStringFromBundle("file.addreplace.success.add");
+
+                            JsonObject successresult = addFileHelper.getSuccessResultAsJsonObjectBuilder().build();
+
+                            try {
+                                logger.fine("successMsg: " + successMsg);
+                                String duplicateWarning = addFileHelper.getDuplicateFileWarning();
+                                if (duplicateWarning != null && !duplicateWarning.isEmpty()) {
+                                    // return ok(addFileHelper.getDuplicateFileWarning(), addFileHelper.getSuccessResultAsJsonObjectBuilder());
+                                    JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                            .add("storageIdentifier ", newStorageIdentifier)
+                                            .add("warning message: ", addFileHelper.getDuplicateFileWarning())
+                                            .add("message ", successresult.getJsonArray("files").getJsonObject(0));
+                                    jarr.add(fileoutput);
+
+                                } else {
+                                    JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                            .add("storageIdentifier ", newStorageIdentifier)
+                                            .add("message ", successresult.getJsonArray("files").getJsonObject(0));
+                                    jarr.add(fileoutput);
+                                }
+
+                            } catch (Exception ex) {
+                                Logger.getLogger(Files.class.getName()).log(Level.SEVERE, null, ex);
+                                return error(Response.Status.BAD_REQUEST, "NoFileException!  Serious Error! See administrator!");
+                            }
+                        }
+
+                        successNumberofFiles = successNumberofFiles + 1;
+                    }
+                }// End of adding files
+            } catch (Exception e) {
+                Logger.getLogger(Files.class.getName()).log(Level.SEVERE, null, e);
```

Comment:
same thing - is there only one type of Exception here/do all of them mean NoFile?